### PR TITLE
Add image optimization to webpack process

### PR DESCRIPTION
**Pull request by Ndpnt**
https://github.com/mxstbr/react-boilerplate/issues/510

_My english is not that good, so feel free to improve the documentation if it could be better :)_

### CODE REVIEW

**Comment by SilentCicero** [comment]
@Ndpnt this looks fairly useful, I think almost everyone would want this as well. Can you explain how much compression this will put on images? Is the amount of compression controllable?

File: docs/general/README.md [hunk 1]
```diff
@@ -114,3 +114,12 @@ with only web-save fonts. In the `body.jsFontLoaded` tag, specify your
 `font-family` stack with your web font.
 
 3. In `app.js` add a `<fontName>Observer` for your font.
+
+## Image optimization
+
+Images often represent the majority of bytes downloaded on a web page, so image
+optimization can often be a notable performance improvement. Thank's to Webpack's
```
Comment (mxstbr) [inline docs/general/README.md]: nit: `Thanks`

File: docs/general/README.md [hunk 2]
```diff
@@ -114,3 +114,12 @@ with only web-save fonts. In the `body.jsFontLoaded` tag, specify your
...
+[`image-loader`](https://github.com/tcoopman/image-webpack-loader), every PNG, JPEG, GIF and SVG images
```
Comment (mxstbr) [inline docs/general/README.md]: nit: `every PNG, JPEG, GIF and SVG image is optimized`

File: docs/general/README.md [hunk 3]
```diff
@@ -114,3 +114,12 @@ with only web-save fonts. In the `body.jsFontLoaded` tag, specify your
...
+are optimized.
+
+See [`image-loader`](https://github.com/tcoopman/image-webpack-loader) to customize optimizations' options.
```
Comment (mxstbr) [inline docs/general/README.md]: nit: no `'` after optimizations

File: docs/general/remove.md [hunk 4]
```diff
@@ -21,3 +21,29 @@ Delete [`manifest.json`](../../app/manifest.json) and remove the
...
+  test: /\.(jpg|png|gif)$/,
+  loaders: [
+    'file-loader',
+    'image-webpack?{progressive:true, optimizationLevel: 7, interlaced: false, pngquant:{quality: "65-90", speed: 4}}',
+  ],
+}
+…
+```
+
+and replace it with classic `file-loader`:
```
Comment (mxstbr) [inline docs/general/remove.md]: nit: `Then replace it`

**Comment by mxstbr** [comment]
This looks very solid, am also a big fan of the documentation! :+1: A few small nitpicks then we'll merge this!

>  Is the amount of compression controllable? 

It even says so in the docs:

> See [`image-loader`](https://github.com/tcoopman/image-webpack-loader) to customize optimizations' options

**Comment by Ndpnt** [comment]
Thanks for your comments :).
All nitpicks are fixed and the branch is rebased onto master.

**Comment by mxstbr** [comment]
Awesome, thanks @Ndpnt! :tada:

**Pull request merged**